### ISSUE TITLE
Add validation hooks on a JetEngine form.

### ISSUE THREAD

**Issue by OverSysDev** [Crocoblock/suggestions#1083]
It would be convenient to have field edit validation hooks apart from the post-submnit actions hook.

It would be very interesting if they included these requirements in a next version of the product.

**Comment by joefclarin** [comment]
I would second this. Recently got into a situation where I need a custom validation of a field. I was actually expecting that it can be done using the "Call a Hook" action but can't find any documentation or examples about it. 

**Why Request Such Feature**
I was working on a client website where two WordPress websites are connected via API. One website is for the admins and the other is for the members. The info of the members are input on the admin website then create the user accounts (using those info) to the members website via API. 

The basic fields to create a user are username and password. Since username is a unique field, I need to check first via API if a username exists or not. If the username exists then the form shouldn't push through the submission and display some error message.

**Comment by arekuki** [comment]
+1

**Comment by igorchernenko92** [comment]
+1

**Comment by sara-kariv** [comment]
MUST HAVE!

**Comment by developernerds** [comment]
Absolutely! I was trying to figure out how to add a filter  so that users can only choose one checkbox value, but can't find any good way to do it and no documentation about it! I really hope they add this feature!

**Comment by hassanaleem0** [comment]
> Absolutely! I was trying to figure out how to add a filter so that users can only choose one checkbox value, but can't find any good way to do it and no documentation about it! I really hope they add this feature!

can't use radio?

**Comment by MjHead** [comment]
You can use hook notification for this purposes. In the JetEngine Forms you need add your validation callback on hook
'jet-engine-booking/filter/your-hook-name' and return true or false depending to validation result

In JetFormBuilder you need to use action 'jet-form-builder/custom-action/your-hook-name' and if you validation was failed, throw new exception with appropriate info:

```throw new \Jet_Form_Builder\Exceptions\Action_Exception( 'your custom message' );```
